### PR TITLE
[Chef] Fix Read/Write Attributes of AirPurifier's clusters. 

### PR DESCRIPTION
**Pull request by erwinpan1**
Fix Read?Write Attributes of AirPurifier's clusters.  Especially that some of these attributes are defined as external and need to write in application

### CODE REVIEW

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp [hunk 1]
```diff
@@ -96,25 +98,161 @@ void HepaFilterMonitoring::Shutdown()
...
+        reader.Init(buffer + sizeof(uint16_t), tlvLen);
+        reader.Next();
+        reader.Get(newValue);
+        DataModel::Nullable<uint32_t> newLastChangedTime = DataModel::MakeNullable(newValue);
+        ret                                              = UpdateLastChangedTime(newLastChangedTime);
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::DegradationDirection::Id:
+    default: {
+        ChipLogError(Zcl, "Unsupported External Attribute Read: %d", static_cast<int>(attributeId));
```
Comment (harimau-qirex) [inline examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp:138]: The log mentions a read but this is in the write method.

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp [hunk 2]
```diff
@@ -96,25 +98,161 @@ void HepaFilterMonitoring::Shutdown()
...
+    case HepaFilterMonitoring::Attributes::InPlaceIndicator::Id: {
+        *(bool *) buffer = GetInPlaceIndicator();
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::LastChangedTime::Id: {
+        DataModel::Nullable<uint32_t> lastChangedTime = GetLastChangedTime();
+        *(uint32_t *) buffer                          = lastChangedTime.IsNull() ? 0 : lastChangedTime.Value();
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::DegradationDirection::Id:
```
Comment (harimau-qirex) [inline examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp:239]: If the condition attribute is supported shouldn't the degradation direction also be supported? They both fall under the CON feature for conformance.

Comment (erwinpan1) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp:239]: @harimau-qirex Actually the limitation comes from here https://source.corp.google.com/search?q=%22%2Fcodelabs%2Fmatter-device-virtual%22&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental . We just support everything except `degradation direction`

```
    Protocols::InteractionModel::Status UpdateCondition(uint8_t aNewCondition);
    Protocols::InteractionModel::Status UpdateChangeIndication(ChangeIndicationEnum aNewChangeIndication);
    Protocols::InteractionModel::Status UpdateInPlaceIndicator(bool aNewInPlaceIndicator);
    Protocols::InteractionModel::Status UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime);
```

**Review by tcarmelveilleux — APPROVED** [review]

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp [hunk 3]
```diff
@@ -96,25 +98,161 @@ void HepaFilterMonitoring::Shutdown()
...
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::InPlaceIndicator::Id: {
+        bool newInPlaceIndicator = *(bool *) buffer;
+        ret                      = UpdateInPlaceIndicator(newInPlaceIndicator);
+    }
+    break;
+    case HepaFilterMonitoring::Attributes::LastChangedTime::Id: {
+        uint32_t newValue = 0;
+        uint16_t tlvLen   = *(uint16_t *) buffer;
```
[the remark this answers is not in the document]
Comment (erwinpan1) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp]: Fixed. PTAL. Thanks.

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp (hunk 1 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp]: Fixed, Thanks for catching this.

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp (hunk 2 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp:250]: Per discussion above, this is the current cluster implementation limitation. Mark this as resolved.

**Review by tcarmelveilleux — APPROVED** [review]

File: examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp (hunk 2 above)
Comment (harimau-qirex) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp:250]: This is reading the attribute, not writing the attribute. The corresponding cluster implementation method is https://github.com/project-chip/connectedhomeip/blob/6842912989bd43a0b15e069eaea33dfc5495de04/src/app/clusters/resource-monitoring-server/resource-monitoring-server.h#L104.

Comment (erwinpan1) [reply examples/chef/common/clusters/resource-monitoring/chef-resource-monitoring-delegates.cpp]: OK. The reading should be supported. Thanks for catching this. The fix is done and committed. So I'll mark this as resolved.

**Pull request merged**